modules/ROOT/pages/hot-deployment.adoc

```diff
@@ -0,0 +1,30 @@
+= Application Hot Deployment
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+:keywords: runtime, deploy, hot deployment, application, applications
+
+Modify your configuration files and custom classes and reload them without having to restart Mule.
+
+== Outline of a Deployment
+
+Here is a quick summary for deploying an app 'foo':
+
+* Create a directory under `$MULE_HOME/apps/foo`
+* Place your packaged application (e.g. `foo.jar`) at `$MULE_HOME/apps`
+* Compress to a jar custom classes (if any), and put them under `$MULE_HOME/apps/foo/META-INF/mule-src/foo/src/main/java/`
+* Start your app with `mule -M-Dmule.deploy.applications=foo`
+
+As a bonus, application's configuration files are monitored, so if there are any class changes you want to pick up or simply modify any config, save or touch any of the Mule configuration files and Mule will hot-reload the application.
+
+== How Hot Deployment works
+
+Mule checks every three seconds for updated configuration files under the `$MULE_HOME/apps` directory, and when it finds one, it reloads the configuration file and the JARs in that applications lib directory. To reload an application, you can:
+
+* Touch the anchor file of that application.
+* Touch or update any of the Mule Configuration Files declared in the xref:intro-packaging#app_descriptor[`mule-artifact.json`] file.
+
+Therefore, if you want to change one of your custom classes, modify it, copy the updated class to the java directory, and then touch the anchor file.
```

Comment:
For example, if you want to modify one of your custom classes, make your changes to the custom class, copy the updated class to the `java` directory, then touch the anchor file.